src/main.rs

```diff
@@ -283,14 +290,14 @@ fn read_ips_from_file(
 /// Given a string, parse it as an host, IP address, or CIDR.
 /// This allows us to pass files as hosts or cidr or IPs easily
 /// Call this everytime you have a possible IP_or_host
-fn parse_to_ip(address: String, resolver: &Resolver) -> Result<Vec<IpAddr>, std::io::Error> {
```

Comment:
This functions signature should also be changed: `address` should be `&str`, because ownership is not needed. The return type should be just the vector as the function always returns `Ok`.